vignettes/query-RESULTS-silviculture-data-using-bcdata.Rmd

```diff
@@ -0,0 +1,215 @@
+
+**How much western larch has been planted in the Prince George Natural Resource District?**  
+
+Let's use the `bcdata` package to search, retrieve and explore the RESULTS silviculture data and answer this question. 
+
+
+### Getting Started
+
+To start, let's load the `bcdata` package. We will also load the `dplyr` and `ggplot2` packages to help us work with the data. You can learn more about the `dplyr` package [here](https://dplyr.tidyverse.org/) and the `ggplot2` package [here](https://ggplot2.tidyverse.org/):
+
+```{r libs, eval = FALSE}
+library(bcdata)
+library(dplyr)
+library(ggplot2)
+```
+
+
+### Getting the Data
+
+We can gather the data we need to answer our question using the [RESULTS - silviculture forest cover dataset](https://catalogue.data.gov.bc.ca/dataset/258bb088-4113-47b1-b568-ce20bd64e3e3). First, let's take a look at the metadata record using `bcdc_get_record()`:
+
+```{r get-record}
+# Get the metadata using the human-readable record name
+bcdc_get_record("results-forest-cover-silviculture")
+```
+
+We see that this is a [Web Service-enabled](https://www2.gov.bc.ca/gov/content?id=95D78D544B244F34B89223EF069DF74E) geospatial data set, so we can query and retrieve this geospatial data set using `bcdc_query_geodata()`:
+
+```{r examine-silv-layer}
+# Query the data using the permanent ID of the record to guard against name changes
+bcdc_query_geodata("258bb088-4113-47b1-b568-ce20bd64e3e3")
+```
+
+This query shows that this data set has over 850,000 features and 159 fields.  Each feature is a treatment unit within a harvested opening, and contains information on the leading five tree species that are present in each treatment unit, including stems per hectare, age, and height.
+
+Note that we have only _queried_ the data set so far---the data set would be too large (~1GB) to download efficiently.  So, let's use `filter` with `bcdc_query_geodata()` to refine our query _before_ we collect the data and import it into R.
+
+
+### Refining a Geospatial Data Query
+
+
+To address our question, we need the treatment data (1) from the Prince George Natural Resource District _and_ (2) that contain western larch.
+
+First, we'll use the `bcdata` package to download the spatial boundary for the Prince George Natural Resource District:
+
+```{r download-dpg}
+## Create a spatial feature object named dpg
+dpg <- bcdc_query_geodata("natural-resource-nr-district") %>%  
+  filter(ORG_UNIT=="DPG") %>% # filter for Prince George Natural Resource District
+  collect() # and collect the data
+```
+
+Let's plot this spatial object and double check we have we what we need:
+
+```{r plot-dpg}
+dpg %>% 
```

Comment:
Would it be permissible to add `dpg` directly into the ggplot2 call and remove the pipe here?